src/test/ui/rfc-2086-irrefutable_let_patterns/deny-irrefutable-if-let.rs

```diff
@@ -0,0 +1,16 @@
+// Copyright 2012 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#[deny(irrefutable_let_patterns)]
+fn main() {
+    if let _ = 5 {
```

Comment:
Put an error code `~ error: irrefutable if-let pattern` or so here so that the test is more robust.